tasks/install.yml

```diff
@@ -0,0 +1,388 @@
+    chdir: "{{ gitlab_install_path }}"
+  async: 31557600 # Wait up to one year for completion
+  poll: 30
+
+- name: Set priviliges on database
+  postgresql_privs:
+    db: "{{gitlab_db_name}}"
+    privs: ALL
+    type: database
+    role: "{{gitlab_db_user}}"
+    unix_socket: "{{postgres_run_directory}}"
+
+- name: Set priviliges on tables
+  postgresql_privs:
+    db: "{{gitlab_db_name}}"
+    privs: ALL
+    type: table
+    objs: ALL_IN_SCHEMA
+    role: "{{gitlab_db_user}}"
+    unix_socket: "{{postgres_run_directory}}"
+
+- name: Set priviliges on sequences
+  postgresql_privs:
+    db: "{{gitlab_db_name}}"
+    privs: ALL
+    type: sequence
+    objs: ALL_IN_SCHEMA
+    role: "{{gitlab_db_user}}"
+    unix_socket: "{{postgres_run_directory}}"
+
+- name: Download systemd service files from gitlab
+  get_url:
+    url:  "https://gitlab.com/gitlab-org/gitlab-recipes/raw/master/init/systemd/{{item}}"
+    dest: "/etc/systemd/system/{{item}}"
+  with_items:
+    - gitlab-sidekiq.service
+    - gitlab-unicorn.service
+    - gitlab-workhorse.service
+    - gitlab-mailroom.service
+    - gitlab-gitaly.service
+
+- name: Enable services
+  systemd:
+    enabled: yes
+    daemon_reload: yes
+    name: "{{item}}"
+  with_items:
+    - gitlab-sidekiq.service
+    - gitlab-unicorn.service
+    - gitlab-workhorse.service
+    - gitlab-mailroom.service
+    - gitlab-gitaly.service
+
+- name: Fetch Gitaly source with git and compile with go
+  command: 'sudo -u git -H bundle exec rake "gitlab:gitaly:install[/home/git/gitaly]" RAILS_ENV=production'
+  args:
+    chdir: "{{ gitlab_install_path }}"
+
+- name: Restrict Gitaly socket access
+  file:
+    path: "{{ gitlab_install_path }}/tmp/sockets/private"
+    owner: git
+    mode: 0700
+
+- name: Setup Logrotate
+  copy:
+    src: "{{ gitlab_install_path }}/lib/support/logrotate/gitlab"
+    dest: /etc/logrotate.d/gitlab
+    remote_src: yes
+
+- name: Compile GetText PO files
+  command: 'sudo -u git -H bundle exec rake gettext:pack RAILS_ENV=production'
+  args:
+    chdir: "{{ gitlab_install_path }}"
+
+- name: Compile GetText PO files
+  command: 'sudo -u git -H bundle exec rake gettext:po_to_json RAILS_ENV=production'
+  args:
+    chdir: "{{ gitlab_install_path }}"
+
+- name: Compile Assets
+  command: 'sudo -u git -H yarn install --production --pure-lockfile'
+  args:
+    chdir: "{{ gitlab_install_path }}"
+
+- name: Compile Assets
+  command: 'sudo -u git -H bundle exec rake gitlab:assets:compile RAILS_ENV=production NODE_ENV=production'
+  args:
+    chdir: "{{ gitlab_install_path }}"
+
+- name: Start GitLab
+  systemd: 
+    state: started
+    name: "{{item}}"
+  with_items:
+    - gitlab-sidekiq.service
+    - gitlab-unicorn.service
+    - gitlab-workhorse.service
+    - gitlab-mailroom.service
+    - gitlab-gitaly.service
+
```

Comment:
This file has 388 lines. Please, for the love of god, split it into smaller task files.
How about
- apt
- postgres
- directories
- plugins
- gitlab_install
- systemd

Reply:
BUMP